crates/viewer/re_space_view_dataframe/src/dataframe_ui.rs

```diff
@@ -28,69 +29,85 @@ pub(crate) enum HideColumnAction {
-        match self {
-            QueryHandle::LatestAt(query_handle) => query_handle.schema(),
-            QueryHandle::Range(query_handle) => query_handle.schema(),
-        }
-    }
+    // The table id mainly drives column widths, so it should be stable across queries leading to
+    // the same schema. However, changing the PoV typically leads to large changes of actual
+    // content. Since that can affect the optimal column width, we include the PoV in the salt.
+    let table_id_salt = egui::Id::new("__dataframe__")
+        .with(&schema)
+        .with(&query_handle.query().filtered_point_of_view);
 
-    fn num_rows(&self) -> u64 {
-        match self {
-            // TODO(#7449): this is in general wrong! However, there is currently no way to know
-            // if the number of row is 0 or 1. For now, we silently accept in the delegate when it
-            // turns out to be 0.
-            QueryHandle::LatestAt(_) => 1,
-            QueryHandle::Range(query_handle) => query_handle.num_rows(),
-        }
-    }
+    // For the row expansion cache, we invalidate more aggressively for now.
+    let row_expansion_id_salt = egui::Id::new("__dataframe_row_exp__")
+        .with(&schema)
+        .with(query_handle.query());
 
-    fn get(&self, start: u64, num_rows: u64) -> Vec<RecordBatch> {
-        match self {
-            QueryHandle::LatestAt(query_handle) => {
-                // latest-at queries only have one row
-                debug_assert_eq!((start, num_rows), (0, 1));
+    let (header_groups, header_entity_paths) = column_groups_for_entity(&schema);
 
-                vec![query_handle.get()]
-            }
-            QueryHandle::Range(query_handle) => query_handle.get(start, num_rows),
-        }
-    }
+    let num_rows = query_handle.num_rows();
 
-    fn timeline(&self) -> Timeline {
-        match self {
-            QueryHandle::LatestAt(query_handle) => query_handle.query().timeline,
-            QueryHandle::Range(query_handle) => query_handle.query().timeline,
-        }
-    }
-}
+    let mut table_delegate = DataframeTableDelegate {
+        ctx,
+        query_handle,
+        schema: &schema,
+        header_entity_paths,
+        num_rows,
+        display_data: Err(anyhow::anyhow!(
+            "No row data, `fetch_columns_and_rows` not called."
+        )),
+        expanded_rows: ExpandedRows::new(
+            ui.ctx().clone(),
+            ui.make_persistent_id(row_expansion_id_salt),
+            expanded_rows_cache,
+            re_ui::DesignTokens::table_line_height(),
+        ),
+        hide_column_actions: vec![],
+    };
 
-impl<'a> From<LatestAtQueryHandle<'a>> for QueryHandle<'a> {
-    fn from(query_handle: LatestAtQueryHandle<'a>) -> Self {
-        QueryHandle::LatestAt(query_handle)
-    }
-}
+    let num_sticky_cols = schema
+        .iter()
+        .take_while(|cd| matches!(cd, ColumnDescriptor::Control(_) | ColumnDescriptor::Time(_)))
```

Comment:
That `take_while` feels fairly risky for very little gain, why not a `filter`?

Reply:
Well we want to have all control columns and time columns to be sticky, but they _must_ appear first in the ~schema~ selected columns.

If something weird like this happened, we'd still want to have only 3 stick column, not 4 (because that would wrongly include `comp1`:

```
row_id  timeA  timeB comp1 comp2 timeC comp3
```